tests/executor/exchange_seq_scan_test.cpp

```diff
@@ -0,0 +1,346 @@
+    executor::LogicalTile *temp = executor.GetOutput();
+    if (result) result->emplace_back(temp);
+  }
+  const auto end = std::chrono::system_clock::now();
+  const std::chrono::duration<double> diff = end - start;
+  const double ms = diff.count() * 1000;
+  return ms;
+}
+
+TEST_F(ExchangeSeqScanTests, LeafNodeCorrectnessTest) {
+  constexpr size_t tile_num = 10;
+  constexpr size_t row_num = 100;
+  // Create table.
+  std::unique_ptr<storage::DataTable> table(CreateTable(tile_num, row_num));
+
+  // Column ids to be added to logical tile after scan.
+  std::vector<oid_t> column_ids({0, 1, 3});
+
+  // Create plan node.
+  planner::SeqScanPlan node(table.get(), CreatePredicate(g_tuple_ids),
+                            column_ids);
+
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  std::unique_ptr<executor::ExecutorContext> context(
+      new executor::ExecutorContext(txn));
+
+  executor::ExchangeSeqScanExecutor executor(&node, context.get());
+  std::vector<executor::LogicalTile *> result;
+  GetRunTime(executor, &result);
+
+  size_t expected_num_tiles = table->GetTileGroupCount();
+  size_t expected_num_cols = column_ids.size();
+
+  EXPECT_EQ(tile_num, expected_num_tiles);
+  EXPECT_EQ(expected_num_tiles, result.size());
+
+  // Check correctness of result tiles.
+  for (size_t i = 0; i < expected_num_tiles; i++) {
+    EXPECT_EQ(expected_num_cols, result[i]->GetColumnCount());
+
+    // Only tuples per tile satisfy our predicate.
+    EXPECT_EQ(g_tuple_ids.size(), result[i]->GetTupleCount());
+
+    // Verify values.
+    std::set<oid_t> expected_tuples_left(g_tuple_ids);
+    for (oid_t new_tuple_id : *(result[i])) {
+      // We divide by 10 because we know how PopulatedValue() computes.
+      // Bad style. Being a bit lazy here...
+
+      int old_tuple_id =
+          result[i]->GetValue(new_tuple_id, 0).GetIntegerForTestsOnly() / 10;
+
+      EXPECT_EQ(1, expected_tuples_left.erase(old_tuple_id));
+
+      int val1 = ExecutorTestsUtil::PopulatedValue(old_tuple_id, 1);
+      EXPECT_EQ(val1,
+                result[i]->GetValue(new_tuple_id, 1).GetIntegerForTestsOnly());
+      int val2 = ExecutorTestsUtil::PopulatedValue(old_tuple_id, 3);
+
+      // expected_num_cols - 1 is a hacky way to ensure that
+      // we are always getting the last column in the original table.
+      // For the tile group test case, it'll be 2 (one column is removed
+      // during the scan as part of the test case).
+      // For the logical tile test case, it'll be 3.
+      Value string_value(ValueFactory::GetStringValue(std::to_string(val2)));
+      EXPECT_EQ(string_value,
+                result[i]->GetValue(new_tuple_id, expected_num_cols - 1));
+    }
+    EXPECT_EQ(0, expected_tuples_left.size());
+  }
+
+  for (auto tile : result) delete tile;
+  txn_manager.CommitTransaction();
+}
+
+TEST_F(ExchangeSeqScanTests, LeafNodeSpeedTest) {
```

Comment:
This test does not actually check any conditions, and relies that you look at output to determine if the result is reasonable. Maybe a check that the parallel version is faster is useful